src/Uno.UI/UI/Xaml/Controls/TimePicker/TimePickerFlyout.iOS.cs

```diff
@@ -1,58 +1,79 @@
 ﻿#if XAMARIN_IOS
+
 using System;
 using System.Collections.Generic;
 using System.Text;
+using Uno.UI;
 using Uno.UI.Common;
+using Uno.UI.Helpers.WinUI;
+using Windows.UI.Xaml.Automation;
 using Windows.UI.Xaml.Controls.Primitives;
 using Windows.UI.Xaml.Data;
 
 namespace Windows.UI.Xaml.Controls
 {
-	//TODO: should inherit from PickerFlyoutBase
 	public partial class TimePickerFlyout : Flyout
 	{
-		private bool _hasAcceptButton = false;
+        private bool AlreadyCancelledOrSaved { get; set; }
 
 		public TimePickerFlyout()
 		{
 			InitializeContent();
 		}
 
-		private void InitializeContent()
+		private void onTap(object sender, Input.PointerRoutedEventArgs e) => e.Handled = true;
+
+		protected internal override void Close()
 		{
-			Content = new TimePickerSelector();
-			BindToContent(nameof(Time));
-			BindToContent(nameof(ClockIdentifier));
+			var timePicker = _presenter as TimePickerFlyoutPresenter;
+			var headerTapZone = timePicker.FindName("HeaderTapZone") as Border;
```

Comment:
Possible `NullReferenceException` here (on `timePicker`)
```suggestion
			var headerTapZone = timePicker?.FindName("HeaderTapZone") as Border;
```

Reply:
Ok i will check for null in these occasions and bypass, hide the bug instead of having an exception that show where the bug is.

personally, for most exceptions of the sort i think that when running in debug the code should fail first on similar exception and you gobble/log exceptions in release in the app and appdomain unhandledexception events

https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.application.unhandledexception

wpf.. 
AppDomain.CurrentDomain.UnhandledException
Dispatcher.UnhandledException 
Application.Current.DispatcherUnhandledException 
TaskScheduler.UnobservedTaskException

Reply:
The headerTapZone is in the template, it is fair for someone using a timepicker to override the template and do one without this headertapzone, the control should keep working and it shouldn't throw an exception.